apps/chat-e2e/src/tests/messageTemplate.test.ts

```diff
@@ -0,0 +1,858 @@
+      .concat(' ')
+      .repeat(10);
+    const truncatedRequestContent = requestContent
+      .substring(0, 160)
+      .concat('...');
+    let conversation: Conversation;
+
+    await dialTest.step('Prepare conversation with long request', async () => {
+      conversation = conversationData.prepareModelConversationBasedOnRequests([
+        requestContent,
+      ]);
+      await dataInjector.createConversations([conversation]);
+    });
+
+    await dialTest.step(
+      'Open "Message Template" modal and verify its content',
+      async () => {
+        await dialHomePage.openHomePage();
+        await dialHomePage.waitForPageLoaded();
+        await conversations.selectConversation(conversation.name);
+        await chatMessages.openMessageTemplateModal(1);
+        await messageTemplateModalAssertion.assertElementText(
+          messageTemplateModal.title,
+          ExpectedConstants.messageTemplateModalTitle,
+        );
+        await messageTemplateModalAssertion.assertElementText(
+          messageTemplateModal.description,
+          ExpectedConstants.messageTemplateModalDescription,
+        );
+        await messageTemplateModalAssertion.assertElementAttribute(
+          messageTemplateModal.getTemplateRowContent(1),
+          Attributes.placeholder,
+          ExpectedConstants.messageTemplateContentPlaceholder,
+        );
+        await messageTemplateModalAssertion.assertElementAttribute(
+          messageTemplateModal.getTemplateRowValue(1),
+          Attributes.placeholder,
+          ExpectedConstants.messageTemplateValuePlaceholder,
+        );
+      },
+    );
+
+    await dialTest.step(
+      'Verify original message is cut with dots',
+      async () => {
+        await messageTemplateModalAssertion.assertElementText(
+          messageTemplateModal.originalMessageContent,
+          truncatedRequestContent,
+        );
+      },
+    );
+
+    await dialTest.step(
+      'Click on "Show more" button and verify full message is displayed',
+      async () => {
+        await messageTemplateModal.showMoreButton.click();
+        await messageTemplateModalAssertion.assertElementText(
+          messageTemplateModal.originalMessageContent,
+          requestContent,
+        );
+        await messageTemplateModal.showLessButton.click();
+        await messageTemplateModalAssertion.assertElementText(
+          messageTemplateModal.originalMessageContent,
+          truncatedRequestContent,
+        );
+      },
+    );
+
+    await dialTest.step(
+      'Set cursor in the first row and verify no new row is added',
+      async () => {
+        await messageTemplateModal.getTemplateRowContent(1).click();
+        await messageTemplateModalAssertion.assertElementsCount(
+          messageTemplateModal.templateRows,
+          1,
+        );
+      },
+    );
+
+    await dialTest.step(
+      'Type request chars in the first row and verify new row is added, delete button is available for the first row',
```

Comment:
Isn't it duplicated by the step "Type request chars in the first row and verify new row is added, delete button is available for the first row"? I understand that we later check if the changes are saved but it can be merged, I assume.

Reply:
agreed, thanks